src/socket.cpp

```diff
@@ -146,13 +146,13 @@ std::string irc::sockets::sockaddrs::addr() const
 	char addrv[INET6_ADDRSTRLEN+1];
 	if (sa.sa_family == AF_INET)
 	{
-		if (!inet_ntop(AF_INET, &in4.sin_addr, addrv, sizeof(addrv)))
+		if (!inet_ntop(AF_INET, (void*)&in4.sin_addr, addrv, sizeof(addrv)))
```

Comment:
Casting to void* here and elsewhere is necessary because inet_ntop takes a PVOID on Windows. This should not affect non-Windows behaviour.